### PR TITLE
Update to ES6 and remove the warnings associated with DeviceEmitter.

### PR DESCRIPTION
**Pull request by sscaff1**
Tested locally. 

DeviceEmitter warning referenced in issue #23

### CODE REVIEW

File: KeyboardSpacer.js [hunk 1]
```diff
@@ -1,89 +1,99 @@
...
+  }
 
-    updateKeyboardSpace(frames) {
-        if (!frames.endCoordinates)
-            return;
+  componentDidMount() {
+    const updateListener = Platform.OS === 'android' ? 'keyboardDidShow' : 'keyboardWillShow';
+    const resetListener = Platform.OS === 'android' ? 'keyboardDidHide' : 'keyboardWillHide';
+    this._listeners = [
+      Keyboard.addListener(updateListener, this.updateKeyboardSpace),
```
Comment (Andr3wHur5t) [inline KeyboardSpacer.js:62]: Does this work with older versions of react native?

File: KeyboardSpacer.js [hunk 2]
```diff
@@ -1,89 +1,99 @@
...
+    this._listeners = null;
+    this.updateKeyboardSpace = this.updateKeyboardSpace.bind(this);
+    this.resetKeyboardSpace = this.resetKeyboardSpace.bind(this);
+  }
 
-    updateKeyboardSpace(frames) {
-        if (!frames.endCoordinates)
-            return;
+  componentDidMount() {
+    const updateListener = Platform.OS === 'android' ? 'keyboardDidShow' : 'keyboardWillShow';
```
Comment (Andr3wHur5t) [inline KeyboardSpacer.js:59]: ~~Have you tested this on android?~~

File: KeyboardSpacer.js [hunk 3]
```diff
@@ -1,89 +1,99 @@
...
-    componentWillUnmount() {
-        this._listeners.forEach(function(/** EmitterSubscription */listener) {
-            listener.remove();
-        });
-    }
+  resetKeyboardSpace() {
+    this.setState({
+      keyboardSpace: 0,
+      isKeyboardOpened: false
+    }, this.props.onToggle(false, 0));
```
Comment (Andr3wHur5t) [inline KeyboardSpacer.js:92]: Might be better to pass in the props directly from the state object.

**Comment by Andr3wHur5t** [comment]
Nice refactor,

I would like to keep this backwards compatible with older versions of react native that don't have the keyboard module.

Other than that looks good.

**Comment by sscaff1** [comment]
@Andr3wHur5t I don't understand. This will work with RN going forward so why not just push a major version bump and then add to the README the appropriate version you should use given a version of RN? I thought this was the standard way of making breaking changes like the above change.

File: KeyboardSpacer.js [hunk 4]
```diff
@@ -1,89 +1,99 @@
 /**
  * Created by andrewhurst on 10/5/15.
  */
-var React = require('react');
-var ReactNative = require('react-native');
+import React, { Component, PropTypes } from 'react';
+import Keyboard from 'Keyboard';
```
Comment (ghuh) [inline KeyboardSpacer.js]: As of RN 0.27.2, this has changed again.  Keyboard should be imported from 'react-native'

**Comment by sscaff1** [comment]
@ghuh thanks - I updated the PR.

**Comment by npomfret** [comment]
Any chance of getting a new release with this fix?

**Comment by mitchellhuang** [comment]
This PR seems to not work when putting the keyboard spacer in a Modal. Not 100% sure but it would be good to double check. I'm running RN 0.27.2.

edit - disregard, i'm dumb

**Comment by Andr3wHur5t** [comment]
Yah, Testing now.

Sorry for latency, been bogged down with work.

**Comment by Andr3wHur5t** [comment]
:ship:

Going to update readme then publish to npm.

**Pull request merged**